Jellyfin.Api/Controllers/UniversalAudioController.cs

```diff
@@ -0,0 +1,412 @@
+        /// <param name="transcodingContainer">Optional. The container to transcode to.</param>
+        /// <param name="transcodingProtocol">Optional. The transcoding protocol.</param>
+        /// <param name="maxAudioSampleRate">Optional. The maximum audio sample rate.</param>
+        /// <param name="maxAudioBitDepth">Optional. The maximum audio bit depth.</param>
+        /// <param name="enableRemoteMedia">Optional. Whether to enable remote media.</param>
+        /// <param name="breakOnNonKeyFrames">Optional. Whether to break on non key frames.</param>
+        /// <param name="enableRedirection">Whether to enable redirection. Defaults to true.</param>
+        /// <response code="200">Audio stream returned.</response>
+        /// <response code="302">Redirected to remote audio stream.</response>
+        /// <returns>A <see cref="Task"/> containing the audio file.</returns>
+        [HttpGet("/Audio/{itemId}/universal")]
+        [HttpGet("/Audio/{itemId}/{universal=universal}.{container?}")]
+        [HttpHead("/Audio/{itemId}/universal")]
+        [HttpHead("/Audio/{itemId}/{universal=universal}.{container?}")]
+        [Authorize(Policy = Policies.DefaultAuthorization)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status302Found)]
+        public async Task<ActionResult> GetUniversalAudioStream(
+            [FromRoute] Guid itemId,
+            [FromRoute] string? container,
+            [FromQuery] string? mediaSourceId,
+            [FromQuery] string? deviceId,
+            [FromQuery] Guid? userId,
+            [FromQuery] string? audioCodec,
+            [FromQuery] int? maxAudioChannels,
+            [FromQuery] int? transcodingAudioChannels,
+            [FromQuery] long? maxStreamingBitrate,
+            [FromQuery] long? startTimeTicks,
+            [FromQuery] string? transcodingContainer,
+            [FromQuery] string? transcodingProtocol,
+            [FromQuery] int? maxAudioSampleRate,
+            [FromQuery] int? maxAudioBitDepth,
+            [FromQuery] bool? enableRemoteMedia,
+            [FromQuery] bool breakOnNonKeyFrames,
+            [FromQuery] bool enableRedirection = true)
+        {
+            bool isHeadRequest = Request.Method == System.Net.WebRequestMethods.Http.Head;
+            var deviceProfile = GetDeviceProfile(container, transcodingContainer, audioCodec, transcodingProtocol, breakOnNonKeyFrames, transcodingAudioChannels, maxAudioSampleRate, maxAudioBitDepth, maxAudioChannels);
+            _authorizationContext.GetAuthorizationInfo(Request).DeviceId = deviceId;
+
+            var mediaInfoController = new MediaInfoController(_mediaSourceManager, _deviceManager, _libraryManager, _networkManager, _mediaEncoder, _userManager, _authorizationContext, _loggerFactory.CreateLogger<MediaInfoController>(), _serverConfigurationManager);
+            var playbackInfoResult = await mediaInfoController.GetPostedPlaybackInfo(
```

Comment:
I know that you're just trying to migrate 1:1, but at least use DI for this.

Reply:
Didn't realize that the controllers could be used through DI. I'm going to have a cleanup PR after everything else is merged; I'll add that to the list

Reply:
When an HTTP request comes in it creates a DI Scope that lives throughout the lifetime of that single request. The controllers are part of that life cycle.